crates/core/src/factories/core_bootstrap.rs

```diff
@@ -0,0 +1,296 @@
+//! The core bootstrap implementation provided by Kitsune2.
+
+use kitsune2_api::{bootstrap::*, config::*, *};
+use std::sync::Arc;
+
+#[cfg(feature = "test_utils")]
+use kitsune2_test_utils::test_bootstrap_srv::TestBootstrapSrv;
+
+const MOD_NAME: &str = "CoreBootstrap";
+
+/// Configuration parameters for [CoreBootstrapFactory].
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CoreBootstrapConfig {
+    /// The url of the kitsune2 bootstrap server. E.g. `https://boot.kitsu.ne`.
+    ///
+    /// # Feature `test_utils`
+    ///
+    /// If the `test_utils` feature is enabled, the default will change.
```

Comment:
I'd quite like to avoid this. As much as possible, features should add new functionality and not change existing functionality. I'm just thinking about later when we want to have a mixture of tests that use the test bootstrapping and others that use a "real" local bootstrap server. We don't be able to do both in the same crate with `test_utils` turned on.

Reply:
This module supports both `test:` bootstrap urls and real actual urls at the same time. It's just the default config that changes when the feature is enabled.

I do agree though that this is a little strange. If you would rather there be clear separate modules for test bootstrapping and production bootstrapping, I'm happy to separate them out.

Reply:
I see the logic at the moment and it makes sense. I'm cautious about it because I'm not sure it will be maintained as cleanly as it's been written and it's one of the patterns that causes me a lot of headache time around Holochain. We have quite a few places where we do things like this that aren't easy to work out with, even if they started out that way.

I'd lean towards separating even if it means a bit of extra code